### ISSUE TITLE
 Inconvenient auto scroll when editing note (reopened)

### ISSUE THREAD

**Issue by taw00** [laurent22/joplin#7617]
This is an extension of  #1546 and a reopened version of #7035

Joplin 2.8.8 Desktop
Platform: Linux x86_64
OS: Fedora 35
Editor: Joplin's plain text editor
Plugins: automatic backlinks, inline tags, link graph UI, quick links, simple backup, templates
Other options: Enable soft breaks is on 
For this test, I disabled any custom userstye.css or userchrome.css

Issue: Joplin jumps around in the editor if you use an imported stylesheet of some complexity. Importantly, this occurs... **even if you are not viewing the web-preview**. I do not have this issue with any other similar editor + previewer editor.

My thoughts on this are that the editor should either be (a) separately scrolled from the preview rendering, or (b) the editor should be the rock solid never move from the cursor winner of any argument between the preview and the editor, and (c) if the preview rendering is not currently displayed, it should not be a factor in the calculations for positioning. I suspect there is a bug here and a feature request.

My test case. I tried to dumb down the example, but it was rather time-consuming and the jumping around a bit is ... something I could not seem to determine what the ultimate trigger was.

So, my test case is rather extravagant. The markdown isn't, but the CSS is. You have asked in the past why folks need to import style sheets. I will give you several reasons.

1. I use the default (or minorly tweaked) userstyles.css for notes and small documents and things. But...
2. I also use Joplin to draft webpages for three sites that accept markdown. I have a stylesheet for each to approximate the styles of those websites. And...
3. I write, a lot. I submit for publication, and I work with critique groups and whatnot. I use the stylesheet in this example to produce a manuscript from my markdown short stories, novel snippets, and whatnot.[^1]

### TL;DR

Attached is a markdown document, `jump-test-for-laurent.md`. Just do this and you should (I hope!) see what I mean.

1. Cut-and-paste the markdown into Joplin
2. Use only the plaintext editor. (I never use the WYSIWYG editor ... for reasons.)
3. Don't open the previewer.
4. Go to the second `</section>` in the document and add a space (with the spacebar) after it. In my instance of Joplin, this consistently causes the viewport to jump towards the beginning of the document.

> Note. The only work-around I found to make the jumping stop is to disable the stylesheet import. I.e., I change `@import` to `@Ximport` for a time.
> 
> And it should be noted that the jumping around gets better and worse. Sometimes switching to a different note and then coming back settles it down. Which is _interesting_.

Attached also is the `log.txt` file. There was nothing in the console.

Thanks for your attention on this.

[^1]: Side note: believe it or not, Joplin may not be built for efficiency (the text is duplicated several times in memory), but it can indeed handle enormous quantities of text just fine. Mostly. I sucked in War and Peace (500K+ words) as an example document and it handled it. It struggled a bit, but it didn't fall over.

[log.txt](https://github.com/laurent22/joplin/files/9932173/log.txt)
[jump-test-for-laurent.md](https://github.com/laurent22/joplin/files/9932174/jump-test-for-laurent.md)

**Comment by laurent22** [comment]
That's very good if you managed to find a way to consistently replicate it, hopefully that will help understand and fix this annoying bug

**Comment by koulikmaity** [comment]
I want to work on this issue, plz assign me this issue

**Comment by roman-r-m** [comment]
> I want to work on this issue, plz assign me this issue

We do not assign issues but feel free to work on it.

**Comment by laurent22** [comment]
> Cut-and-paste the markdown into Joplin
> Use only the plaintext editor. (I never use the WYSIWYG editor ... for reasons.)
> Don't open the previewer.
> Go to the second </section> in the document and add a space (with the spacebar) after it. In my instance of Joplin, this consistently causes the viewport to jump towards the beginning of the document.

Yes that's very good, it does it every single time with these steps. Doesn't mean I'll know how to fix it, but at least we have something to work with

**Comment by laurent22** [comment]
Well I'm happy I found and fixed a subtle bug in @ken1kob's otherwise excellent code. It definitely fixes the bug in this particular issue and doesn't seem to have broken scrolling in regular notes. Hopefully that will also fixes the random "scroll to top" bug that's been around for some times

**Comment by joshpark8** [comment]
I'm having a similar issue on version 2.9.17, revision: a84a8e7. I've found it happens when I use backspace while part of my left palm is resting on my trackpad, but I haven't been able to recreate it 100% of the time. I'm on a MacBook Pro M2 running MacOS Ventura 13.2.

**Comment by taw00** [comment]
> Well I'm happy I found and fixed a subtle bug in @ken1kob's otherwise excellent code

We have been sitting on 2.9.17 for some time now (since November of last year). It is possible to solidify a 2.10 version so we can get this rather annoying bug behind us? Thanks.

**Comment by taw00** [comment]
> We have been sitting on 2.9.17 for some time now (since November of last year). It is possible to solidify a 2.10 version so we can get this rather annoying bug behind us? Thanks.

So, Laurent, in the discourse chat stated that the newer versions 2.10.z are ready for prime time, he just hasn't had time to put together the release notes to formalize the process.

I tested the new builds and indeed, it looks like this incredibly annoying problem is likely behind us. I could shed a tear. Thank you dev team!

**Comment by black-A** [comment]
version：2.11.9
still auto random scrolling when edit in editor，maddening！

**Comment by taw00** [comment]
> version：2.11.9 still auto random scrolling when edit in editor，maddening！

Yeah. I still see it … sometimes. I can't reproduce it reliably just yet. It's 1000% better than what it once was, but indeed, it is still a nagging little problem.

**Comment by pschmidtke** [comment]
Still seeing this also rather regularly only in markdown editing mode without preview on MacOs Ventura 13.6.1
and 
```
Joplin 2.13.8 (prod, darwin)

Client ID: 4ee966d7d19b4022880c7ad465795cd3
Sync Version: 3
Profile Version: 44
Keychain Supported: Yes

Revision: 604dcbc
```
